apps/meteor/client/components/UserAutoCompleteMultiple/UserAutoCompleteMultiple.tsx

```diff
@@ -1,55 +1,125 @@
+
 	const { data } = useQuery({
-		queryKey: ['usersAutoComplete', debouncedFilter],
-		queryFn: async () => usersAutoCompleteEndpoint(query(debouncedFilter)),
+		queryKey: usersQueryKeys.userAutoComplete(debouncedFilter, federated ?? false),
+
+		queryFn: async () => {
+			const users = await getUsers({ selector: JSON.stringify({ term: debouncedFilter }) });
+			const options = users.items.map((item): [string, UserAutoCompleteOptionType] => [item.username, item]);
+
+			// Add extra option if filter text matches `username:server`
+			// Used to add federated users that do not exist yet
+			if (federated && matrixRegex.test(debouncedFilter)) {
+				options.unshift([debouncedFilter, { name: debouncedFilter, username: debouncedFilter, _federated: true }]);
+			}
+
+			return options;
+		},
+
+		placeholderData: keepPreviousData,
 	});
 
-	const options = useMemo(() => data?.items.map((user) => ({ value: user.username, label: user.name })) || [], [data]);
+	const options = useMemo(() => data || [], [data]);
+
+	const onAddUser = useCallback(
+		(username: string): void => {
+			const user = options?.find(([val]) => val === username)?.[1];
+			if (!user) {
+				throw new Error('UserAutoCompleteMultiple - onAddSelected - failed to cache option');
+			}
+			setSelectedCache((selectedCache) => ({ ...selectedCache, [username]: user }));
+		},
+		[setSelectedCache, options],
+	);
+
+	const onRemoveUser = useCallback(
+		(username: string): void =>
+			setSelectedCache((selectedCache) => {
+				const users = { ...selectedCache };
+				delete users[username];
+				return users;
+			}),
+		[setSelectedCache],
+	);
+
+	const handleOnChange = useCallback(
+		(usernames: string[]) => {
+			onChange(usernames);
+			const newAddedUsername = usernames.filter((username) => !value?.includes(username))[0];
+			const removedUsername = value?.filter((username) => !usernames.includes(username))[0];
+			setFilter('');
+			newAddedUsername && onAddUser(newAddedUsername);
+			removedUsername && onRemoveUser(removedUsername);
+		},
+		[onChange, setFilter, onAddUser, onRemoveUser, value],
+	);
 
 	return (
-		<AutoComplete
-			{...props}
-			filter={filter}
-			setFilter={setFilter}
-			onChange={onChange}
-			multiple
-			renderSelected={({ selected: { value: username, label }, onRemove, ...props }): ReactElement => (
-				<UserAvatarChip {...props} username={username} name={label} mie={4} onClick={onRemove} />
-			)}
-			renderItem={({ value, label, ...props }): ReactElement => (
-				<Option data-qa-type='autocomplete-user-option' key={value} {...props}>
-					<OptionAvatar>
-						<UserAvatar username={value} size='x20' />
-					</OptionAvatar>
-					<OptionContent>
-						{label} <OptionDescription>({value})</OptionDescription>
-					</OptionContent>
-				</Option>
-			)}
-			options={options}
-		/>
+		<OptionsContext.Provider value={{ options }}>
+			<MultiSelectFiltered
+				{...props}
+				data-qa='create-channel-users-autocomplete'
```

Comment:
Are these qa properties needed? If so this component is used on places other than create channel.